internal/provider/template_test.go

```diff
@@ -0,0 +1,43 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```

Comment:
It's a long standing position of HC to try to avoid dependency that are not required.

Please use `go-cmp` that is already part of the codebase for these tests.

Reply:
it is done